core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -507,7 +508,17 @@ public void process(@Nullable MappingMetaData mappingMd, String concreteIndex) {
         if (id == null) {
             assert autoGeneratedTimestamp == -1 : "timestamp has already been generated!";
             autoGeneratedTimestamp = Math.max(0, System.currentTimeMillis()); // extra paranoia
-            id(UUIDs.base64UUID());
+            String uid;
+            if (indexCreatedVersion == null) {
+                // we don't know the creation version yet, so default to random uuids, which are the only safe choice
```

Comment:
This is a part that I wanted to discuss with someone who better understands implicit index creation when the index of an index request does not exist yet. Is there a way to know what its created version will be?

The issue is that I could not change the uid generation algorithm in place as there would be a rare chance that uids would collide. So I'm looking for ways to make things work when the index is not created. Using a random uid should work, but I wonder whether we could do better.